com.sap.cloud.lm.sl.cf.process/src/main/java/com/sap/cloud/lm/sl/cf/process/util/ClientReleaser.java

```diff
@@ -15,23 +15,23 @@
 
     private static final Logger LOGGER = LoggerFactory.getLogger(ClientReleaser.class);
 
-    private final FlowableEngineEvent event;
     private final CloudControllerClientProvider clientProvider;
+    private final HistoryService historyService = Context.getProcessEngineConfiguration()
+            .getHistoryService();
 
-    public ClientReleaser(FlowableEngineEvent event, CloudControllerClientProvider clientProvider) {
-        this.event = event;
+    public ClientReleaser(CloudControllerClientProvider clientProvider) {
         this.clientProvider = clientProvider;
     }
 
-    public void releaseClient() {
-        HistoryService historyService = Context.getProcessEngineConfiguration()
-            .getHistoryService();
-        String processInstanceId = event.getProcessInstanceId();
+    public void releaseClient(FlowableEngineEvent event) {

```

Comment:
Not good as the releaseClient is taking event???
Maybe the only below method should be left and this to be deleted becase it will be consistent.

Reply:
Also, try to rename it to releaseClientFor or similar.